packages/playground/src/pages/GiftCards/GiftCards.js

```diff
@@ -72,18 +72,41 @@ import '../../style.scss';
         }
     });
 
+    const checkoutButton = document.querySelector('#custom-checkout-button');
+
+    const giftcardCheckBalance = () => window.giftcard.balanceCheck();
+    const giftcardSubmit = () => window.giftcard.submit();
+    const cardSubmit = () => window.card.submit();
+
+    checkoutButton.addEventListener('click', giftcardCheckBalance);
+
+    const setupGiftCardSubmit = () => {
+        checkoutButton.removeEventListener('click', giftcardCheckBalance);
+        checkoutButton.addEventListener('click', giftcardSubmit);
+    };
+    const setupCardSubmit = () => {
+        checkoutButton.removeEventListener('click', setupGiftCardSubmit);
+        checkoutButton.addEventListener('click', cardSubmit);
+    };
+
     window.giftcard = sessionCheckout
         .create('giftcard', {
             type: 'giftcard',
             brand: 'svs',
-            onOrderCreated: (data) => {
-                afterGiftCard(data);
+            onSubmit: () => {
+                afterGiftCard();
+            },
+            onOrderCreated: () => {
+                afterGiftCard();
+            },
+            onRequiringConfirmation: () => {
```

Comment:
Maybe we should pass the Giftcard balance as part of the `onRequiringConfirmation` callback,  so the merchant can do the same thing that we do on our Drop-in implementation: show the giftcard balance and show the remaining amount.  

Does it make sense or am I missing something?

Reply:
I think this could make sense, at the moment we already display the amount in the giftcard component itself.